src/hidimstat/conditional_feature_importance.py

```diff
@@ -124,10 +126,11 @@ def fit(self, X, y=None, groups=None, var_type="auto"):
                     if self.imputation_model_categorical is None
                     else clone(self.imputation_model_categorical)
                 ),
-                random_state=self.random_state,
+                # require a RandomState due to scikitlearn check
+                random_state=[group_id, seed_root],
```

Comment:
How does that work? it seems to me that `check_random_state` does not support list imputs

Reply:
I add the option in `check_random_state` for including the possibilities of list inputs.

Reply:
Then it will not allow reproducible results for the reason discussed above and in #360 

IMO, we should stay with sklearn's seeding practices, and if we change the check_random_state function, it means we deviate from this.

Reply:
Yes, but I can't accept bad practices on random generators in parallel environments.